src/pages/Explore/TraceExploration.tsx

```diff
@@ -333,11 +334,44 @@ const EmbeddedHeader = ({ model }: SceneComponentProps<TraceExplorationScene>) =
   const { returnToPreviousSource } = traceExploration.useState();
   const filtersVariable = getFiltersVariable(traceExploration);
   const primarySignalVariable = getPrimarySignalVariable(traceExploration);
+  const metricVariable = traceExploration.getMetricVariable();
   const timeRangeControl = traceExploration.state.controls.find((control) => control instanceof SceneTimePicker);
+  
+  const [explorationUrl, setExplorationUrl] = React.useState(() => getUrlForExploration(traceExploration));
 
   // Force the primary signal to be 'All Spans'
   primarySignalVariable?.changeValueTo(primarySignalOptions[1].value!);
 
+  useEffect(() => {
```

Comment:
nit: I think the code might be a bit simple if subscriptions are in the scene object (with `._subs` you get automatic clean up on deactivation) + having `explorationUrl` in scene's state (then we can just use `model.useState()` instead of having `useEffect`)